user_tools/src/spark_rapids_pytools/cloud_api/sp_types.py

```diff
@@ -727,10 +729,11 @@ def _set_remaining_configuration_list(self) -> None:
                                                         sectionKey=self.ctxt.get(config_file_section[config_file]))
                 if loaded_conf_dict:
                     self.ctxt.update(loaded_conf_dict)
-            for prop_elem in properties_map_arr:
-                if loaded_conf_dict and prop_elem.get('propKey') not in loaded_conf_dict:
+            for prop_entry in properties_map_arr:
+                prop_entry_key = prop_entry.get('propKey')
+                if self.ctxt.get(prop_entry_key) is None:
```

Comment:
Can we add a comment to explain what we are trying to do?
I know that the original code did not have much explanation, but it will be nice to add that since we are already modifying the code.

Reply:
Added comment. PTAL